### PR TITLE
adding cdn-base for uploadcare-php

### PR DESCRIPTION
**Pull request by merrymaker14**

### CODE REVIEW

File: src/Uploadcare/Api.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Api
    *
    * @var string
    */
-  public $cdn_host = 'ucarecdn.com';
+  private $cdn_host = 'ucarecdn.com';
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Api.php]: why

File: src/Uploadcare/Api.php [hunk 2]
```diff
@@ -118,11 +118,10 @@ public function __construct($public_key, $secret_key, $userAgentName = null, $cd
   {
     $this->public_key = $public_key;
     $this->secret_key = $secret_key;
+	$this->cdn_host = $cdn_host;
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Api.php]: tabs?

Comment (dmitry-mukhin) [inline src/Uploadcare/Api.php]: why

File: src/Uploadcare/Api.php [hunk 3]
```diff
@@ -118,11 +118,10 @@ public function __construct($public_key, $secret_key, $userAgentName = null, $cd
...
     $this->widget = new Widget($this);
     $this->uploader = new Uploader($this);
-    if($cdn_host !== null) {
-      $this->cdn_host = $cdn_host;
-    }
```
Comment (dmitry-mukhin) [inline src/Uploadcare/Api.php]: why

File: src/Uploadcare/File.php [hunk 4]
```diff
@@ -168,6 +168,18 @@ public function delete()
...
+  
+  /**
+   * Get url path of original file
+   *
+   * @param int $post_id
+   * @return string
+   */
+  public function getPath($post_id)
+  {
+    return $uploadcare_path = get_post_meta($post_id, 'uploadcare_path', true);
```
Comment (dmitry-mukhin) [inline src/Uploadcare/File.php]: wtf

File: src/Uploadcare/File.php [hunk 5]
```diff
@@ -168,19 +168,20 @@ public function delete()
...
   {
-    $url = sprintf('%s/%s/', $this->api->getCdnUri(), $this->uuid);
-    if($this->default_effects) {
-      $url = sprintf('%s-/%s', $url, $this->default_effects);
+    $uuid = $this->uuid;
+    if ($this->default_effects) {
+      $path = sprintf('%s/%s/', $uuid, $this->default_effects);
     }
-    if($this->filename && $postfix === null) {
+	if($this->filename && $postfix === null) {
```
Comment (dmitry-mukhin) [inline src/Uploadcare/File.php]: wham

File: src/Uploadcare/File.php [hunk 6]
```diff
@@ -168,19 +168,20 @@ public function delete()
...
    */
-  public function getUrl($postfix = null)
+  public function getPath($postfix = null)
   {
-    $url = sprintf('%s/%s/', $this->api->getCdnUri(), $this->uuid);
-    if($this->default_effects) {
-      $url = sprintf('%s-/%s', $url, $this->default_effects);
+    $uuid = $this->uuid;
+    if ($this->default_effects) {
+      $path = sprintf('%s/%s/', $uuid, $this->default_effects);
```
Comment (dmitry-mukhin) [inline src/Uploadcare/File.php]: this results in incorrect path
https://uploadcare.com/documentation/cdn/#file-api

File: src/Uploadcare/File.php [hunk 7]
```diff
@@ -227,11 +228,21 @@ public function getUrl($postfix = null)
...
   }
+  
+  /**
+   * Get url of original image
+   *
+   * @return string
+   */
+  public function getUrl($postfix = null)
+  {
+    $url = sprintf('%s/%s/', $this->api->getCdnUri(), $this->getPath());
```
Comment (dmitry-mukhin) [inline src/Uploadcare/File.php]: you've list the `$postfix`

**Comment by ZNick1982** [comment]
Modifications moved to the https://github.com/uploadcare/uploadcare-php/pull/92 pr

**Pull request closed without merging**